staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1013,18 +952,12 @@ func (e *Store) Delete(ctx genericapirequest.Context, name string, options *meta
 
 	// Handle combinations of graceful deletion and finalization by issuing
 	// the correct updates.
-	if e.EnableGarbageCollection {
-		shouldUpdateFinalizers, _ := deletionFinalizers(e, accessor, options)
-		// TODO: remove the check, because we support no-op updates now.
-		if graceful || pendingFinalizers || shouldUpdateFinalizers {
-			err, ignoreNotFound, deleteImmediately, out, lastExisting = e.updateForGracefulDeletionAndFinalizers(ctx, name, key, options, preconditions, obj)
-		}
-	} else {
-		// TODO: remove the check on graceful, because we support no-op updates now.
-		if graceful {
-			err, ignoreNotFound, deleteImmediately, out, lastExisting = e.updateForGracefulDeletion(ctx, name, key, options, preconditions, obj)
```

Comment:
We used to keep both functions to be able to turn off all finalizers. The finalizers feature has been stablized long enough to remove the redundant code.